access/pagerduty/pagerduty_test.go

```diff
@@ -303,6 +314,31 @@ func (s *PagerdutySuite) TestDenial(c *C) {
 	c.Assert(request.GetState(), Equals, services.RequestState_DENIED)
 }
 
+func (s *PagerdutySuite) TestAutoApprovalWhenOnCall(c *C) {
+	s.appConfig.Pagerduty.AutoApprove = true
+	s.userName = s.me.Username + "@example.com"
+	s.fakePagerduty.AddOnCall(pd.OnCall{})
```

Comment:
@awly i thought it's an overkill to implement a real OnCall filtering by multiple parameters in `FakePagerduty`. And this approach works well since in my code I just check `len(onCalls) > 0` and trust that Pagerduty API works correctly doesn't return bad data such an empty `pd.OnCall{}`.

Reply:
If we rely on third-party API correctness all the time why we shouldn't trust it in this case?

Anyway, I'll write that double check of onCall data so I have to fix this one too.

Reply:
This is a sensitive piece of code, since it enables privilege escalation.
I'd prefer to be more paranoid about all the 3rd party responses.